linux-setup/readme.txt

```diff
@@ -0,0 +1,4 @@
+You'll need to setcap the lokinet to make sure you don't have to run it as root
+On debian-based distros, you make sure you have setcap installed first: apt install libcap2-bin
+and then you can:
+setcap cap_net_admin,cap_net_bind_service=+eip lokinet
```

Comment:
We aren't doing this anymore for the debs because it breaks some setups (such as having a NFS root); instead we add capabilities in the service file.

Reply:
I'm aware, this text was created in late December when I was told I shouldn't add a shell script for installation, that a text file was preferred. My goal here is to get it included into the Repo, so we can all make edits and improve it (not be a file that's only on my containers). We can argue over the contents thru PRs that attempt to make further edits to it.

My personal thought is for those that do have setcap, this seems like a better flow. And not all distros have systemd. I could see including some systemd service files in `contrib/` and also providing some instructions in this file on how to install and use those.

Reply:
Okay, here's some suggested wording:

---
Lokinet needs certain capabilities to run to set up a virtual network interface and provide a DNS server.  The preferred approach to using this is through the linux capabilities mechanism, which allows assigning limited capabilities without needing to run the entire process as root.

There are two main ways to do this:

1. If you are running lokinet via an init system such as systemd, you can specify the capabilities in the service file by adding:

CapabilityBoundingSet=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
AmbientCapabilities=CAP_NET_ADMIN CAP_NET_BIND_SERVICE

into the `[Service]` section of the systemd service file.  This will assign the necessary permissions when running the process and allow lokinet to work while running as a non-root user.

2. You can set the capabilities on the binary by using the `setcap` program (if not available you may need to install `libcap2-bin` on Debian/Ubuntu-based systems) and running:

setcap cap_net_admin,cap_net_bind_service=+eip lokinet

This grants the permissions whenever the `lokinet` binary is executed.